EpiAware/src/EpiLatentModels/CombineLatentModels.jl

```diff
@@ -0,0 +1,62 @@
+@doc raw"
+The `CombineLatentModels` struct.
+
+This struct is used to combine multiple latent models into a single latent model.
+
+# Constructors
+
+- `CombineLatentModels(models::M) where {M <: AbstractVector{<:AbstractTuringLatentModel}}`: Constructs a `CombineLatentModels` instance with specified models, ensuring that there are at least two models.
+- `CombineLatentModels(; models::M) where {M <: AbstractVector{<:AbstractTuringLatentModel}}`: Constructs a `CombineLatentModels` instance with specified models, ensuring that there are at least two models.
+
+# Examples
+
+```julia
+using EpiAware, Distributions
+combined_model = CombineLatentModels([Intercept(Normal(2, 0.2)), AR()])
+latent_model = generate_latent(combined_model, 10)
+latent_model()
+```
+"
+@kwdef struct CombineLatentModels{M <: AbstractVector{<:AbstractTuringLatentModel}} <:
+              AbstractTuringLatentModel
+    "A vector of latent models"
+    models::M
+
+    function CombineLatentModels(models::M) where {M <:
+                                                   AbstractVector{<:AbstractTuringLatentModel}}
+        @assert length(models)>1 "At least two models are required"
+        return new{AbstractVector{<:AbstractTuringLatentModel}}(models)
+    end
+end
+
+@doc raw"
+Generate latent variables using a combination of multiple latent models.
+
+# Arguments
+- `latent_models::CombineLatentModels`: An instance of the `CombineLatentModels` type representing the collection of latent models.
+- `n`: The number of latent variables to generate.
+
+# Returns
+- `combined_latents`: The combined latent variables generated from all the models.
+- `latent_aux`: A tuple containing the auxiliary latent variables generated from each individual model.
+
+# Example
+"
+@model function EpiAwareBase.generate_latent(latent_models::CombineLatentModels, n)
+    @submodel final_latent, latent_aux = _accumulate_latents(
+        latent_models.models, 1, fill(0.0, n), [], n, length(latent_models.models))
+
+    return final_latent, (; latent_aux...)
+end
+
+@model function _accumulate_latents(models, index, acc_latent, acc_aux, n, n_models)
+    if index > n_models
```

Comment:
This adds to the number of functions being generated with conditionals; which means that if we use `ReverseDiff` as our AD method we'll have to restrict to uncompiled/re-taped calculations.

I'm not 100% against that, this is how stan at least used to go (e.g. [here](https://arxiv.org/pdf/1509.07164) but it might well have moved forward), but I think its worth reflecting upon whether we can mitigate with performance annotations (c.f. [here](https://juliadiff.org/ReverseDiff.jl/dev/api/#Optimization-Annotations)).

Reply:
I thought the conditional issue was only for dynamic quantities?

Reply:
Gotcha. This is ok.

Reply:
I'm not 100% on this FYI!

Reply:
I'm thinking!